addons/website_slides/models/slide_slide.py

```diff
@@ -700,165 +816,231 @@ def _compute_quiz_info(self, target_partner, quiz_done=False):
     # Parsing methods
     # --------------------------------------------------
 
-    @api.model
-    def _fetch_data(self, base_url, params, content_type=False):
-        result = {'values': dict()}
+    def _fetch_external_metadata(self, fetch_image=True, return_when_not_found=False):
```

Comment:
return_when_not_found is strangely named. Especially it changes the output of sub-methods ... don't really like it but hey. Maybe a renaming would ease understanding, like ``allow_return_error`` ? ``return_void_on_not_found=True`` ?

Reply:
Counter attack: make sub methods return two things (values, error), and handle only values here and error when needed ? Better than a ping-pong style parameter.